ethcore/node-filter/src/lib.rs

```diff
@@ -37,28 +37,44 @@ extern crate tempdir;
 #[macro_use]
 extern crate log;
 
+use std::collections::{HashMap, VecDeque};
 use std::sync::Weak;
 
-use ethcore::client::{BlockChainClient, BlockId};
+use ethcore::client::{BlockChainClient, BlockId, ChainNotify, NewBlocks};
+
 use ethereum_types::{H256, Address};
 use ethabi::FunctionOutputDecoder;
 use network::{ConnectionFilter, ConnectionDirection};
 use devp2p::NodeId;
+use devp2p::MAX_NODES_IN_TABLE;
+use parking_lot::RwLock;
 
 use_contract!(peer_set, "res/peer_set.json");
 
 /// Connection filter that uses a contract to manage permissions.
 pub struct NodeFilter {
 	client: Weak<BlockChainClient>,
 	contract_address: Address,
+	cache: RwLock<Cache>
```

Comment:
If we understand the real reason for CPU overload (not limited Node table), can we just rollback #10143 and return previous LRUCache, that was used here? I frankly don't see the advantages of proposed solution in comparison with the previous one.

Reply:
The idea of implementing it via `RwLock` was to make it possible to read the cache from different threads without blocking each other.

Reply:
Ok. Than may be just change previous Mutex<LruCache> to RwLock<LruCache>? Because order field in Cache is just the implementation of Lru

Reply:
Got it! I think that makes sense. Sorry I didn't notice and I was thinking that `LruCache` comes already with with built-in `Mutex` 😅

Reply:
Actually I cannot really use `LruCache` since it does not provide non-mutable `get` method, so it's not possible to use `RwLock` with it

Reply:
What about the `peek` method: https://docs.rs/lru/0.1.13/lru/struct.LruCache.html#method.peek ?